src/utils/graph/graphUtils.ts

```diff
@@ -0,0 +1,34 @@
+
+// types.ts
+export interface XYPoint {
+  x: number;
+  y: number;
+}
+
+export interface Segment {
+  name: string; // Assuming segments have a name
+  value: number;
+}
+
+export type GraphData = XYPoint[] | Segment[];
+
+
+
+export class GraphUtils {
```

Comment:
missed this on the first review, but also this file already exists see:
`/src/utils/graph/graph` which has funcitons for isSegmentData and `isXYData` respectively.

Please use these functions rather than creating new ones.
See comment below for getMin/getMax/getTotal